lib/index.js

```diff
@@ -37,47 +37,50 @@ module.exports = {
 		var msgs = {}, msgid = 1, handlers = {}, processMsg, smtpServer, that;
 		smtpServer = smtp.createServer(smtpOptions);
 		smtpServer.on("startData",function(envelope){
-		  envelope.body = "";
+			envelope.body = "";
 		});
 		smtpServer.on("data",function(envelope,chunk){
-		  envelope.body += chunk;
+			envelope.body += chunk;
 		});
 		smtpServer.on('authorizeUser', function(connection, username, password, callback) {
 			callback(null, true);
 		});
 
 		smtpServer.on("dataReady",function(envelope,callback){
-			var email = {sender: null, receivers: {}, data: ""}, mp = new MailParser();
+			var email = {sender: null, receivers: {}, data: ""};
 			email.sender = envelope.from;
 			_.each(envelope.to||[],function(rcpt){
 				/*jslint regexp:true */
 				email.receivers[rcpt.replace(/^[^<]*</g,"").replace(/>[^>]*$/g,"")] = true;
 				/*jslint regexp:false */
 			});
 			email.data = envelope.body;
-			mp.on("end",function (mail_object) {
-				_.extend(email,mail_object || {});
-				// handle the http://tools.ietf.org/html/rfc2821#section-4.5.2 double-dot at the beginning of a line
-				// thanks to http://github.com/andris9 for pointing it out
-				//email.body = email.data.replace(/^\.\./mg,'.');
-				// is this email encoded quotedprintable? decode it
-				/*
-				if (email.headers && email.headers["Content-Transfer-Encoding"] && email.headers["Content-Transfer-Encoding"].toLowerCase() === "quoted-printable") {
-					email.body = mimelib.decodeQuotedPrintable(email.body);
+			// Handle the http://tools.ietf.org/html/rfc2821#section-4.5.2 double-dot at the beginning of a line
+			var unescapedBody = envelope.body.replace(/^\.\./mg, '.');
+			simpleParser(unescapedBody, function(err, mail_object) {
+				if (err)
+					callback(err);
+				else {
+					_.extend(email,mail_object || {});
+					// handle the http://tools.ietf.org/html/rfc2821#section-4.5.2 double-dot at the beginning of a line
+					// email.body = email.text.replace(/^\.\./mg, '.');
+					// is this email encoded quotedprintable? decode it
+					/*
+					if (email.headers && email.headers["Content-Transfer-Encoding"] && email.headers["Content-Transfer-Encoding"].toLowerCase() === "quoted-printable") {
+						email.body = mimelib.decodeQuotedPrintable(email.body);
+					}
+					*/
+
+					// legacy support
+					email.headers.To = email.to.text;
+					email.headers.From = email.from.text;
+					email.body = email.text;
```

Comment:
I assumed those were hard-coded for tests? Didn't make sense to have them in here.